apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -1,14 +1,55 @@
 import type { Request } from 'express';
 import _ from 'lodash';
+import fetch, { RequestInfo, RequestInit } from 'node-fetch';
+import { Issuer, TokenSet } from 'openid-client';
 import { z } from 'zod';
 
 import { HttpStatusError } from '@prairielearn/error';
-import { loadSqlEquiv, queryRow } from '@prairielearn/postgres';
+import { loadSqlEquiv, queryRow, queryAsync, runInTransactionAsync } from '@prairielearn/postgres';
 
+import {
+  DateFromISOString,
+  Lti13InstanceSchema,
+  Lti13CourseInstanceSchema,
+} from '../../lib/db-types.js';
 import { features } from '../../lib/features/index.js';
+import { ServerJob } from '../../lib/server-jobs.js';
+import { selectLti13Instance } from '../models/lti13Instance.js';
 
 const sql = loadSqlEquiv(import.meta.url);
 
+const TOKEN_SCOPES = [
+  'https://purl.imsglobal.org/spec/lti-ags/scope/lineitem',
+  //'https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly',
+  'https://purl.imsglobal.org/spec/lti-ags/scope/score',
+  //'https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly',
+  //'https://purl.imsglobal.org/spec/lti-ags/scope/lineitem.readonly',
+  //'https://canvas.instructure.com/lti/public_jwk/scope/update',
```

Comment:
The non-commented lines are the scope we currently need to fulfill the functionality so far. We'll likely have to do something with contextmembership.readonly in the next part, so I left the full list in to make it easier to add them later (if we discover we need them). Or I can remove them.

Reply:
Changed in 6cb43ef48

Reply:
Hmm, why not include all the scopes we reasonably think we'd need in the near future? That seems preferable to having to go back to every LTI admin to request additional scopes in the future (I imagine that's what we'd have to do otherwise, right?).